[WIP] Revive project Part 2: new SDK

Agrael1 opened the pull request:
This part will try to refactor code to a new SDK.
Accomplishments:

1. Move to community SDK with async interfaces
2. Add Pipelines for automatic publishing
3. Separate shared publishing for VS2019 and VS2022

The PR is under dev.

Commit:
Separation, multiple targets, new SDK

Commit:
IWYU complete async with build pipeline

Commit:
Refactored format includes

Commit:
buildable proto

Commit:
Functional downloader have 2 fix IWYU

Commit:
fixed loader

Commit:
Restored all functionality

Commit:
Cleanup, ready for merge

Commit:
updated readme

Commit:
version

Agrael1 commented:
The project is functional, however, the graph window is removed, as it requires a lot more work to be done. 
The VSIX is built by two separate projects, that use different code for VS2019 and VS2022
Publishing pipelines will be in the 3rd part with test coverage

Commit:
fix for property value

Commit:
Fixed substitution
added options for smart and cheap substitution
Fixed commentary option
Added a class for smart formatting, allowing substitution extension

Commit:
added formatting option after IWYU

Commit:
added selection IWYU for project context menu

Commit:
fixed IWYU for files

Commit:
fixed visibility

Commit:
added include minimizer, fixed the Uri resolution for numerous subprojects
Mapping of includes for IWYU!

Commit:
all IWYUs are now downloading application

Commit:
fixed multiple selection IWYU

Commit:
Added parser and final precise mode for includes and fwd decls, added moving decls, todo:
Make empty namespace dissappear

Commit:
empty namespace removal done

Commit:
fixed file parse, tested on real project

Commit:
little underscore fix

Commit:
cleanup

Agrael1 committed:
Create main.yml

Agrael1 committed:
Update main.yml

Commit:
output window renamed

Agrael1 committed:
Update main.yml

Wumpf commented:
@Agrael1 haven't looked at all the things in detail but I get the impression everything is in good hands here. Can I DM you somehow to figure out how to do a handover of the VS marketplace part so you can take full control over the plugin and publish new versions going forward?

Agrael1 commented:
@Wumpf of course
maybe discord is a great way, Agrael#7152

Wumpf commented:
yeah, but where and how? :D
Twitter/Discord/Telegram/Signal/etc.? Github still doesn't have a system for DMs :/
Twitter might be easiest cause my handle is already public (don't want to post phonenumbers etc. more than necessary) https://twitter.com/wumpf/

Agrael1 committed:
Create vs-publish.json

Agrael1 committed:
Update README.md

Agrael1 committed:
Update README.md

Commit:
ready for publish

Commit:
little fixes to the .md

Commit:
version revert

Agrael1 committed:
Merge branch 'master' into Workflow

Agrael1 committed:
[release] first release

Create main.yml

Agrael1 committed:
[release] fixed pipeline

Agrael1 committed:
[release] split artifacts

Commit:
[release] fixes for pipeline

Agrael1 committed:
fixed schema

Agrael1 committed:
pipeline fix

Commit:
new formatter, using more efficient parser, further decrease of DTE

Commit:
Merge branch 'master' of https://github.com/Agrael1/IncludeToolbox

Commit:
bad habits

Commit:
tests will be later as I remove DTE dependencies and make better classes

Commit:
removed tests because of dependency resolution inconsistensies

Commit:
quick version fix
fixed IWYU formatting usage

Commit:
newline fixes

Agrael1 committed:
Update main.yml

Agrael1 committed:
)

Agrael1 committed:
[release] release with new feature

Agrael1 committed:
Update main.yml

Agrael1 committed:
[release] final touches

Agrael1 committed:
fixed internal name

Commit:
tag short

Agrael1 committed:
[release] pipeline fix

Commit:
[release] vsix schema

Commit:
Merge branch 'master' of https://github.com/Agrael1/IncludeToolbox

Agrael1 committed:
[release] fixed readme

Commit:
moved Header span to utilities, added additional ReadOnlySpan version

Commit:
Merge branch 'master' of https://github.com/Agrael1/IncludeToolbox

Commit:
[release]Refactored trial and error
Added parser tests
Added Testing to the pipeline steps

Commit:
fixed version

Commit:
fixes to pipeline

Commit:
Added property for default msvc mappings

Commit:
fixed images
Added Project Wide Trial and Error remover

Commit:
Fixed IWYU deadly parser bugs (no newline on the end of the file)
Added parser tests.
Replaces Include with IncludeLine
Moved data structures to its own file
IWYU now only affects first file parsed.

Commit:
fixed NSTracker

Agrael1 committed:
[release] fixes to the IWYU parser

Fixed IWYU deadly parser bugs (no newline on the end of the file)

Commit:
Added preproc support with pragma once

Commit:
Fixed formatting and IWYU
Added brains to precise mode, so it won't add anything to preprocessor sections.
Fixed Empty namespaces
Added tests

Commit:
cleanup and changes

Commit:
missing icon

Agrael1 committed:
[release] Bugfixes and Precise feature

Added preproc support with pragma once

Commit:
Enabled Trial and Error removal for header files, wrote a little guide

Agrael1 committed:
[release] TAEIR for headers

Enabled Trial and Error removal for header files, wrote a little guide

Commit:
Added precompile header file support for Trial and Error

Agrael1 committed:
[release] pch for headers Trial and Error fix

Added precompile header file support for Trial and Error

Commit:
reinsertion bug

Agrael1 committed:
[release]bugfix

reinsertion bug

Commit:
fixed IWYU throw on empty addition

Agrael1 committed:
[release] fixed IWYU

fixed IWYU throw on empty addition

Commit:
[release] removed VS Publish step, publishing will be manual, updated version

Commit:
Updated readme with links, reverted VSIX ID for VS2019

Commit:
[release] include graph beta

Commit:
[release] fix structs

Commit:
[release] added relative path extension

Commit:
[release] forgot to save

Commit:
[release] another attempt at publishing

Commit:
[release] CI/CD pipeline

Commit:
[release] filetypes

Commit:
fixed UE NMake include format

Agrael1 committed:
Merge pull request #10 from Agrael1/FixFormatUE

fixed UE NMake include format

Commit:
[release] fix for UE

Commit:
Merge branch 'master' of https://github.com/Agrael1/IncludeToolbox

Commit:
[release] workflow

Commit:
[release] compile header tool

Commit:
[release] fix mapper for empty headers

dako98 committed:
Fix spaces in path.

Fix call to IWYU when the path to the file has spaces.

Agrael1 committed:
Merge pull request #19 from dako98/master

Fix calling IWYU with spaces in the path
